tests/cupy_tests/manipulation_tests/test_kind.py

```diff
@@ -10,6 +10,19 @@
 @testing.gpu
 class TestKind(unittest.TestCase):
 
+    @testing.for_all_dtypes()
+    def test_asarray_chkfinite(self, dtype):
+        a = [0, 4, 0, 5]
+        a_gpu = cupy.asarray_chkfinite(a, dtype=dtype)
+        a_cpu = numpy.asarray_chkfinite(a, dtype=dtype)
+        assert a_gpu.dtype == a_cpu.dtype
+
+    @unittest.expectedFailure
+    @testing.for_all_dtypes()
+    def test_asarray_chkfinite_non_finite_vals(self, dtype):
+        a = [-numpy.inf, 0., numpy.inf]
+        self.assertRaises(ValueError, cupy.asarray_chkfinite, a, dtype=dtype)
```

Comment:
See https://github.com/cupy/cupy/pull/3256 for how to check exceptions.
```suggestion
        for xp in (numpy, cupy):
            with pytest.raises(ValueError):
                xp.asarray_chkfinite(a, dtype=dtype, order=order)
```

Reply:
> See #3256 for how to check exceptions.

Thanks for sharing the informative PR. I read that thoroughly (: 
I was wondering if a bit of enhancement can be made in code. For example, here, arrays are created inside the `for-loops`; that is not needed as it is independent of `xp`.
```
def test_dim_mismatch_ellipsis2(self):
        for xp in (numpy, cupy):
            a = xp.arange(12).reshape(2, 3, 2)
            with pytest.raises(ValueError):
                xp.einsum('i...,...i', a, a)
```
We can modify it to:
```
def test_dim_mismatch_ellipsis2(self):
        a = xp.arange(12).reshape(2, 3, 2)
        for xp in (numpy, cupy):
            with pytest.raises(ValueError):
                xp.einsum('i...,...i', a, a)
```
#### Gains
- Will take lesser computation time.

What do you think? Thanks for reading it (:

Reply:
No, `xp` is the loop variable, so it can't be pulled out of the loop.

Reply:
Oops, agreed. I was walking without looking into the other side :(